### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by Elhambasir**
### What is done in this PR?
In this project, I added unit tests to the codebase, specifically to the classes.
- [x] Added unit tests for all the methods in all the classes:

- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Rental`

[PROJECT REQUIREMENTS](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md)

Note: I don't know why it is throwing this RSpec error in Github I tried a lot but could not fix that. But it's working properly on my local machine you can check the attached file here:
school-2-for-reviewer

Thank you for reviewing :)

### CODE REVIEW

**Review by franclobo — COMMENTED** [review]
Hi @Elhambasir and @LDouglasOT, 

# Invalid Code Review Request

descarga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors ❌ 

```ruby
An error occurred while loading ./spec/classroom_spec.rb. - Did you mean?
                    rspec ./spec/decorator_spec.rb

Failure/Error: require_relative '../classroom'

LoadError:
  cannot load such file -- /home/runner/work/add-person-student-and-teacher-classes/add-person-student-and-teacher-classes/Person
# ./student.rb:1:in `require_relative'
# ./student.rb:1:in `<top (required)>'
# ./classroom.rb:1:in `require_relative'
# ./classroom.rb:1:in `<top (required)>'
# ./spec/classroom_spec.rb:1:in `require_relative'
# ./spec/classroom_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/student_spec.rb. - Did you mean?
                    rspec ./spec/rental_spec.rb

Failure/Error: require_relative '../student'

LoadError:
  cannot load such file -- /home/runner/work/add-person-student-and-teacher-classes/add-person-student-and-teacher-classes/Person
# ./student.rb:1:in `require_relative'
# ./student.rb:1:in `<top (required)>'
# ./spec/student_spec.rb:1:in `require_relative'
# ./spec/student_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/teacher_spec.rb. - Did you mean?
                    rspec ./spec/rental_spec.rb

Failure/Error: require_relative '../teacher'

LoadError:
  cannot load such file -- /home/runner/work/add-person-student-and-teacher-classes/add-person-student-and-teacher-classes/Person
# ./teacher.rb:1:in `require_relative'
# ./teacher.rb:1:in `<top (required)>'
# ./spec/teacher_spec.rb:1:in `require_relative'
# ./spec/teacher_spec.rb:1:in `<top (required)>'

Finished in 0.00004 seconds (files took 0.205[8](https://github.com/Elhambasir/add-person-student-and-teacher-classes/actions/runs/5485631298/jobs/9994673998?pr=8#step:5:9)8 seconds to load)
0 examples, 0 failures, 3 errors occurred outside of examples

Error: Process completed with exit code 1.
```

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    expect(book.author).to eq('elham')
+  end
+
+  it 'Book must has a rental' do
+    expect(book.rental).to eq([])
+  end
+
+  it 'Book must has an id' do
+    expect(book.id).to be_a(Integer)
+  end
```
Comment (danmainah) [inline spec/book_spec.rb:25]: - [x] @Elhambasir  and @LDouglasOT Great work adding tests for your book class however you forgot to add tests for the `add_rental` method.

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    expect(@student.rental).to eq([])
+  end
+
+  it 'has a classroom' do
+    expect(@student.classroom).to eq('A')
+  end
+
+  it 'can not use services (under age no parent permission)' do
+    expect(@student.can_use_services?).to eq(true)
+  end
```
Comment (danmainah) [inline spec/student_spec.rb:34]: - [x] @team Great work adding tests for the student class, however, you forgot to add tests for the method `play_hooky`.

File: spec/classroom_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  let(:student1) { Student.new('Elham', 23, classroom) }
+  let(:student2) { Student.new('John', 24, classroom) }
+
+  it 'should have a label' do
+    expect(classroom.label).to eq('CS101')
+  end
+
+  it 'should have an empty list of students' do
+    expect(classroom.students).to be_empty
+  end
```
Comment (danmainah) [inline spec/classroom_spec.rb:23]: - [x] @team Great work adding tests for the classroom class, however, you forgot to add tests for the `add_student` method,kindly consider adding tests for it.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Elhambasir and @LDouglasOT , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Ensured all tests added are passing.
✔️ Professional written README file with all sections required.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Elhambasir** [comment]
Ok thank you

**Review by OLIPLICHE — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Added unit tests (✅)) 👏
- All test has passed👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**